bioio_ome_tiled_tiff/reader.py

```diff
@@ -44,66 +63,184 @@ class Reader(BaseReader):
+                return False
+
+            with BioReader(path, backend="python") as br:
+                # Fail fast if multi-image file
+                if len(br.metadata.images) > 1:
+                    raise exceptions.UnsupportedFileFormatError(
+                        path,
+                        "This file contains more than one scene and only the first "
+                        + "scene can be read by the OmeTiledTiffReader. "
+                        + "To read additional scenes, use the TiffReader, "
+                        + "OmeTiffReader, or BioformatsReader.",
+                    )
+
+                return True
+
+        # tifffile exceptions
+        except (TypeError, ValueError):
+            return False
+
+    def __init__(
+        self,
+        image: types.PathLike,
+        chunk_dims: Optional[Union[str, List[str]]] = None,
+        out_order: str = dimensions.DEFAULT_DIMENSION_ORDER,
+        fs_kwargs: Dict[str, Any] = {},
+        **kwargs: Any,
+    ):
+        # Expand details of provided image
+        self._fs, self._path = io.pathlike_to_fs(
+            image,
+            enforce_exists=True,
+            fs_kwargs=fs_kwargs,
+        )
+
+        if not isinstance(self._fs, LocalFileSystem):
+            raise ValueError(
+                "Cannot read .ome.tif from non-local file system. "
+                f"Received URI: {self._path}, which points to {type(self._fs)}."
+            )
+
+        try:
+            self._rdr = BioReader(self._path, backend=self.backend)
+        except (TypeError, ValueError, TiffFileError):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+        # Add ndim attribute so _rdr can be passed directly to dask
+        self._rdr.ndim = len(self._rdr.shape)
+
+        # Setup dimension ordering
+        dims = "YXZCT"
+        self.native_dim_order = dims[: len(self._rdr.shape)]
+        assert all(d in out_order for d in dims)
+        self.out_dim_order = [d for d in out_order if d in dims]
+
+        # Currently do not support custom chunking, throw a warning.
+        if chunk_dims is not None:
+            log.warning(
+                "OmeTiledTiffReader does not currently support custom chunking."
+            )
 
     @property
     def scenes(self) -> Tuple[str, ...]:
-        """
-        Return the list of available scenes for the file using the
-        cached parameters stored to the object in the __init__.
-        """
-        raise NotImplementedError()
+        return tuple(image_meta.id for image_meta in self._rdr.metadata.images)
 
-    def _read_delayed(self) -> "xr.DataArray":
-        """
-        Return an xarray DataArray filled with a delayed dask array, coordinate planes,
-        and any metadata stored in the attrs.
+    @property
+    def current_scene(self) -> str:
+        return self.scenes[self._current_scene_index]
```

Comment:
and this

Reply:
resolved in [3e46353](https://github.com/bioio-devs/bioio-ome-tiled-tiff/pull/2/commits/3e46353b361a2b17d4edfec8399cafba3914344b)